tests/components/totalconnect/test_config_flow.py

```diff
@@ -0,0 +1,79 @@
+"""Tests for the iCloud config flow."""
+# from unittest.mock import MagicMock
+
+from homeassistant import data_entry_flow
+from homeassistant.components.totalconnect.const import DOMAIN
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from tests.common import MockConfigEntry
+
+USERNAME = "username@me.com"
+USERNAME_2 = "second_username@icloud.com"
+PASSWORD = "password"
+
+
+async def test_user(hass):
+    """Test user config."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data=None
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+    # test with all provided
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD},
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
```

Comment:
This test looks to allow the TotalConnect client to do I/O, that should be mocked

Reply:
think this is complete.